### PR TITLE
fix: fix inconsistent plan when agent data resources run during apply…

### PR DESCRIPTION
**Pull request by pedro-te**
… always.

Addresses:
> │ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for module.thousandeyes-tests.thousandeyes_http_server.Server-test-ire[0] to include new values learned so far during apply, provider "registry.terraform.io/thousandeyes/thousandeyes" produced an invalid new value for
│ .agents: planned set element cty.ObjectVal(map[string]cty.Value{"agent_id":cty.UnknownVal(cty.Number), "agent_name":cty.StringVal(""), "agent_state":cty.StringVal(""), "agent_type":cty.UnknownVal(cty.String),
│ "cluster_members":cty.ListValEmpty(cty.Object(map[string]cty.Type{"agent_state":cty.String, "ip_addresses":cty.List(cty.String), "last_seen":cty.String, "member_id":cty.Number, "name":cty.String, "network":cty.String, "prefix":cty.String,
│ "public_ip_addresses":cty.List(cty.String), "target_for_tests":cty.String, "utilization":cty.Number})), "country_id":cty.StringVal(""), "created_date":cty.StringVal(""), "enabled":cty.NullVal(cty.Bool),
│ "error_details":cty.ListValEmpty(cty.Object(map[string]cty.Type{"code":cty.String, "description":cty.String})), "groups":cty.SetValEmpty(cty.Object(map[string]cty.Type{"builtin":cty.Bool, "group_id":cty.Number, "name":cty.String,
│ "type":cty.String})), "hostname":cty.StringVal(""), "ip_addresses":cty.UnknownVal(cty.List(cty.String)), "ipv6_policy":cty.StringVal(""), "keep_browser_cache":cty.NullVal(cty.Bool), "last_seen":cty.StringVal(""), "location":cty.StringVal(""),
│ "network":cty.StringVal(""), "prefix":cty.StringVal(""), "target_for_tests":cty.StringVal(""), "utilization":cty.NullVal(cty.Number), "verify_ssl_certificate":cty.NullVal(cty.Bool)}) does not correlate with any element in actual.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵

I can provide an explanation on why this fixes the issue via slack if needed. But essentially, during the apply phase, terraform compares what it had in the plan with what was learned during apply, but it was unable to compare the agents as per the error message:
> (...) does not correlate with any element in actual.

This is due to the fact that it's trying to match these two, I think:
This is what is present in the test resource
```
              {
                "agent_id": 56299,
                "agent_name": "",
                "agent_state": "",
                "agent_type": "",
                "cluster_members": [],
                "country_id": "",
                "created_date": "",
                "enabled": false,
                "error_details": [],
                "groups": [],
                "hostname": "",
                "ip_addresses": [],
                "ipv6_policy": "",
                "keep_browser_cache": false,
                "last_seen": "",
                "location": "",
                "network": "",
                "prefix": "",
                "target_for_tests": "",
                "utilization": 0,
                "verify_ssl_certificate": false
              }
```

and this is what it learns during the apply for the data resource:
```
{
            "agent_id": 56299,
            "agent_name": "st-381-aws-thousandeyes-agent",
            "agent_state": null,
            "agent_type": null,
            "cluster_members": null,
            "country_id": null,
            "created_date": null,
            "enabled": null,
            "error_details": null,
            "groups": null,
            "hostname": null,
            "id": "0x1400071bf38",
            "ip_addresses": null,
            "ipv6_policy": null,
            "keep_browser_cache": null,
            "last_seen": null,
            "location": null,
            "network": null,
            "prefix": null,
            "target_for_tests": null,
            "utilization": null,
            "verify_ssl_certificate": null
          }
```

I think all those nulls being compared to the empty strings were causing the issue. Since only `agent_id` matters here, since everything else is removed here https://github.com/thousandeyes/terraform-provider-thousandeyes/blob/main/thousandeyes/util.go#L164 , I removed all other fields, so now only `agent_id` is compared, which is always set.

### CODE REVIEW

File: thousandeyes/data_source_thousandeyes_agent.go [hunk 1]
```diff
@@ -14,20 +14,15 @@ func dataSourceThousandeyesAgent() *schema.Resource {
...
-				Computed: true,
+				Type:        schema.TypeInt,
+				Computed:    true,
 				Description: "The unique ID of the agent.",
 			},
-			"agent_type": {
-				Type:     schema.TypeString,
-				Computed: true,
-				Description: "The type of agent.",
-			},
```
Comment (pedro-te) [inline thousandeyes/data_source_thousandeyes_agent.go:30]: Not needed, this gets cleared here: https://github.com/thousandeyes/terraform-provider-thousandeyes/blob/main/thousandeyes/util.go#L164

File: thousandeyes/schemas.go [hunk 2]
```diff
@@ -54,224 +54,6 @@ var schemas = map[string]*schema.Schema{
 				"agent_id": {
 					Type:        schema.TypeInt,
 					Description: "The unique ID for the ThousandEyes agent.",
-					Optional:    true,
```
Comment (pedro-te) [inline thousandeyes/schemas.go:57]: These fields are not needed, all of them, as they get removed here: https://github.com/thousandeyes/terraform-provider-thousandeyes/blob/main/thousandeyes/util.go#L164

**Comment by brumarqu-te** [comment]
@pedro-te i think `ipAddress` matters in agent tests (check `agents.sourceIpAddress` in the [test metadata](https://developer.thousandeyes.com/v6/tests/#/test_metadata)) but that's for a different PR i guess, since we were already stripping out all the fields and just using the agent_id.

**Review by brumarqu-te — APPROVED** [review]

**Comment by pedro-te** [comment]
> test metadata

Yeah that's the thing, it seems we never supported it. But we can add this functionality in a different PR for sure. 👍🏻

**Review by sfreitas-te — APPROVED** [review]

**Pull request merged**